docs/guides/directing-traffic-to-your-app.md

```diff
@@ -0,0 +1,375 @@
+    Message:               Gateway is accepted
+    Observed Generation:   1
+    Reason:                Accepted
+    Status:                True
+    Type:                  Accepted
+    Last Transition Time:  2023-08-15T20:57:21Z
+    Message:               Gateway is programmed
+    Observed Generation:   1
+    Reason:                Programmed
+    Status:                True
+    Type:                  Programmed
+  Listeners:
+    Attached Routes:  1
+    Conditions:
+      Last Transition Time:  2023-08-15T20:57:21Z
+      Message:               Listener is accepted
+      Observed Generation:   1
+      Reason:                Accepted
+      Status:                True
+      Type:                  Accepted
+      Last Transition Time:  2023-08-15T20:57:21Z
+      Message:               Listener is programmed
+      Observed Generation:   1
+      Reason:                Programmed
+      Status:                True
+      Type:                  Programmed
+      Last Transition Time:  2023-08-15T20:57:21Z
+      Message:               All references are resolved
+      Observed Generation:   1
+      Reason:                ResolvedRefs
+      Status:                True
+      Type:                  ResolvedRefs
+      Last Transition Time:  2023-08-15T20:57:21Z
+      Message:               No conflicts
+      Observed Generation:   1
+      Reason:                NoConflicts
+      Status:                False
+      Type:                  Conflicted
+    Name:                    http
+  ```
+
+  Check that the conditions match and that the attached routes for the `http` listener equals 1. If it is 0, there may
+  be an issue with the HTTPRoute.
+
+- Check the status of the HTTPRoute:
+
+  ```shell
+  kubectl describe httproute coffee
+  ```
+
+  The HTTPRoute status should look similar to this:
+
+  ```text
+  Status:
+    Parents:
+      Conditions:
+        Last Transition Time:  2023-08-15T20:57:21Z
+        Message:               The route is accepted
+        Observed Generation:   1
+        Reason:                Accepted
+        Status:                True
+        Type:                  Accepted
+        Last Transition Time:  2023-08-15T20:57:21Z
+        Message:               All references are resolved
+        Observed Generation:   1
+        Reason:                ResolvedRefs
+        Status:                True
+        Type:                  ResolvedRefs
+      Controller Name:         gateway.nginx.org/nginx-gateway-controller
+      Parent Ref:
+        Group:      gateway.networking.k8s.io
+        Kind:       Gateway
+        Name:       cafe
+        Namespace:  default
+  ```
+
+  Check for any error messages in the conditions.
+
+- Check the generated nginx config:
+
+  ```shell
+  kubectl exec -it -n nginx-gateway <nginx gateway Pod> -c nginx -- nginx -T
+  ```
+
+  The config should contain the following config blocks:
+
+  ```nginx configuration
+  upstream default_coffee_80 {
```

Comment:
I expect this will change as we evolve NKG generated config. I wonder if there is a good way to prevent the information here from getting outdated?

Reply:
Good point. I'll rework this and see if I can future-proof it.

Reply:
Reworked, let me know what you think.

I can also remove this bullet point if it's not helpful or will be too hard to maintain.